addons/arrays/test_find.sqf

```diff
@@ -0,0 +1,185 @@
+// ----------------------------------------------------------------------------
+#define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+SCRIPT(test_find);
+
+// ----------------------------------------------------------------------------
+
+private ["_original", "_expected", "_result", "_fn"];
+
+_fn = "CBA_fnc_findNil";
+LOG("Testing " + _fn);
+
+TEST_DEFINED("CBA_fnc_findNil","");
+
+// Find First Nil Entry
+_original = ["", player, "", player, nil, "", player, nil];
+_result = ["", player, "", player, nil, "", player, nil] call CBA_fnc_findNil;
+_expected = 4;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
```

Comment:
The three tests below all do the same thing as the first test above - look for the first literal `nil` element in the test arrays. Keep the first one, and change the three redundant tests to use
 * an array with a variable in it somewhere that is nil
 * an array with a code element in it that returns nil
 * an empty input array ([])